src/Node/Expression/BlockReferenceExpression.php

```diff
@@ -40,9 +40,19 @@ public function compile(Compiler $compiler): void
             if ($this->getAttribute('output')) {
                 $compiler->addDebugInfo($this);
 
-                $this
-                    ->compileTemplateCall($compiler, 'displayBlock')
-                    ->raw(";\n");
+                if ($compiler->getEnvironment()->useYield()) {
+                    $compiler->write('yield from ');
+                }
+
+                if ($compiler->getEnvironment()->useYield()) {
```

Comment:
why using 2 conditions here ?

Reply:
The joy of moving things around :) Fixed now.